packages/core/src/Cardano/types/StakePool/APExtendedStakePoolMetadata.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Based on:
+ * https://a.adapools.org/extended-example
```

Comment:
That was the initial idea of having just one common interface to expose pool extended metadata but it turned out the two formats are incompatible and not all properties are optional.

An example is `pool.contact.primary` from CIP-6 [format](https://raw.githubusercontent.com/cardano-foundation/CIPs/master/CIP-0006/schema.json) is a required one, however, it's missing from AP format so it opens an issue with the mapping to a common ext metadata interface, which is the reason we choose to support both formats.

I put that one in the core package since it is just a sibling of the other CIP-6 interface. @mkazlauskas

Reply:
| could you please explain why https://github.com/input-output-hk/cardano-js-sdk/commit/9e7d7ced0713df8b6d9ab8d06758414e0e5ec403 was needed
Because we introduce new [HttpMetadataService](https://github.com/input-output-hk/cardano-js-sdk/blob/c4eafd09c3d8bac4b0bdbc17849533e5c9a6fede/packages/cardano-services/src/Metadata/types.ts#L7-L11) interface, it's a HTTP one in contrast of the existing `MetadataService` that's why I saw reasonable to add the prefix to distinguish easily between both interfaces (HTTP and DbSync). I decided to not use one service with two methods since if someone needs to use only HTTP then there is no need to provide database dependency as well.

Reply:
If we are not obligated to return "one or another" ext data format then we can define and expose our own common `StakePoolExtendedMetadata` with all fields nullable. Probably worth to be discussed with the Lace team because they have to reconcile this variation, thoughts? @rhyslbw  @mkazlauskas

Reply:
What is the alternative though? I really don't like having AP types in core package, and we don't want to butcher CIP6 type to fit AP either, cause then it won't be accurate to the CIP.

Reply:
Good points. The lack of adoption with CIP-6 probably makes this an obvious direction.

Reply:
Also, if we leave it as-is, Lace will have to handle AP metadata with optional fields anyway, so it's just as difficult on their side either way.

Reply:
I'll discuss this with the Lace team to get their input, not a problem. Making our own common type makes the most sense to me as we'll reach a greater amount of users. I can also raise the lack of CIP-6 adoption in the CIP forum and get that conversation started.